build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.code-quality.gradle.kts

```diff
@@ -80,89 +53,71 @@ dependencies {
-                    nullawayEnabled = errorproneExtension.nullawayEnabled
-                    nullawayAwareDeps = configurations.named(compileClasspathConfigurationName).map {
-                        it.incoming.artifacts
-                    }
-                }
-            }
-        }
-
-        @Suppress("UnstableApiUsage")
-        fun addErrorProneDependency(dep: String) {
-            project.dependencies.addProvider(
-                annotationProcessorConfigurationName,
-                extension.enabled.filter { it }.map { dep }
-            )
-        }
-
-        // don't forget to update the version in distributions-dependencies/build.gradle.kts
-        addErrorProneDependency("com.google.errorprone:error_prone_core:2.42.0")
-        addErrorProneDependency("com.uber.nullaway:nullaway:0.12.10")
-
-        project.tasks.named<JavaCompile>(this.compileJavaTaskName) {
+        project.tasks.named<JavaCompile>(sourceSet.compileJavaTaskName) {
             options.errorprone {
                 isEnabled = extension.enabled
-                checks = errorproneExtension.disabledChecks.map {
-                    it.associateWith { CheckSeverity.OFF }
-                }
-
                 nullaway {
-                    checkContracts = true
-                    isJSpecifyMode = true
-                    severity = errorproneExtension.nullawayEnabled.map { if (it) CheckSeverity.ERROR else CheckSeverity.OFF }
+                    severity = errorproneExtension.nullawayEnabled.map { if (it) ERROR else OFF }
                 }
             }
         }
+        // don't forget to update the version in distributions-dependencies/build.gradle.kts
+        addErrorProneDependency(sourceSet.annotationProcessorConfigurationName, extension, "com.google.errorprone:error_prone_core:2.42.0")
+        addErrorProneDependency(sourceSet.annotationProcessorConfigurationName, extension, "com.uber.nullaway:nullaway:0.12.10")
     }
 }
 
 tasks.withType<JavaCompile>().configureEach {
     options.errorprone {
+        disableAllWarnings = true // considering this immense spam burden, remove this once to fix dedicated flaw. https://github.com/diffplug/spotless/pull/2766
         disableWarningsInGeneratedCode = true
-        allErrorsAsWarnings = true
+        disable("JavaxInjectOnAbstractMethod") // We use abstract injection as a pattern
+        error(
+            // TBD:
+            // "EffectivelyPrivate", // It is still useful to distinguish between public interface and implementation details of inner classes even though it isn't enforced.
+            // "EnumOrdinal", // This violation is ubiquitous, though most are benign.
+            // "EqualsGetClass", // Let's agree if we want to adopt Error Prone's idea of valid equals()
+            // "JdkObsolete", // Most of the checks are good, but we do not want to replace all LinkedLists without a good reason
+            "MissingOverride",
+            "SelfAssignment",
+            "StringCharset",
```

Comment:
this build should fail, according to:

- https://github.com/gradle/gradle/pull/35871
- https://github.com/google/error-prone/issues/5378